### PR TITLE
chore: add tests to `EnumCoder`

### PR DESCRIPTION
**Pull request by msensys**
This PR creates tests to ensure that some validations are `validating` correctly.

- Added validation to ensure only one field is provided when encoding enums
- Added validation to ensure decoded data has the appropriate size

PS: With these tests, we achieved 100% coverage

### CODE REVIEW

**Comment by msensys** [comment]
hey @arboleya, could you take a look? The `Test/e2e` failed

**Comment by danielbate** [comment]
Hey @msensys, thankyou for your contribution :) Please could you sign the CLA?

**Comment by msensys** [comment]
@danielbate, I think that all is right

File: .changeset/rotten-walls-tickle.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@fuel-ts/abi-coder": minor
```
Comment (danielbate) [inline .changeset/rotten-walls-tickle.md:2]:
```suggestion
```

Test changes don't require a package redeployment as it is not shipped with the build.

**Comment by danielbate** [comment]
#2546 should resolve the problem you currently have in CI

**Review by danielbate — APPROVED** [review]

**Review by maschad — APPROVED** [review]
Thanks for your contribution! @msensys

**Pull request merged**